src/app/(auth)/signup/page.tsx

```diff
@@ -14,8 +14,28 @@ export default function SignUp() {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [emailSentCount, setEmailSentCount] = useState(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
   // const { push } = useRouter();
   const handleSignUp = async () => {
+    if (email === '' || password === '') {
```

Comment:
i don't think we need this bigger if statement if we're checking them again..

Reply:
i have this statement so that it returns out of the func before calling the supabase signup function in case either email or password is invalid.